ovs-service/src/main/java/org/dcsa/ovs/service/VesselScheduleService.java

```diff
@@ -1,21 +1,172 @@
 package org.dcsa.ovs.service;
 
+import lombok.Builder;
 import lombok.RequiredArgsConstructor;
-import org.dcsa.ovs.mapping.VesselScheduleMapper;
-import org.dcsa.ovs.persistence.repository.VesselScheduleRepository;
-import org.dcsa.ovs.mapping.ListMapper;
-import org.dcsa.ovs.transferobjects.VesselScheduleTO;
+import org.dcsa.ovs.persistence.entity.*;
+import org.dcsa.ovs.persistence.repository.ServiceRepository;
+import org.dcsa.ovs.transferobjects.*;
+import org.dcsa.ovs.transferobjects.enums.DimensionUnit;
+import org.dcsa.ovs.transferobjects.enums.PortCallStatusCode;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.Objects;
 
 @Service
 @RequiredArgsConstructor
 public class VesselScheduleService {
-  private final VesselScheduleRepository repository;
-  private final VesselScheduleMapper mapper;
 
-  public List<VesselScheduleTO> findAll() {
-    return ListMapper.convertList(repository.findAll(), mapper::toTO);
+  private final ServiceRepository serviceRepository;
+
+  @Builder
+  public static class ServiceSchedulesFilters {
+    String carrierServiceCode;
+    String universalServiceReference;
+    String vesselIMONumber;
+    String vesselName;
+    String voyageNumber;
+    String universalVoyageReference;
+    String unLocationCode;
+    String facilitySMDGCode;
+    String startDate;
+    String endDate;
+    Integer limit;
+    String cursor;
+    String apiVersion;
```

Comment:
What do we use this one for? (I cannot find it used anywhere)

Reply:
The idea was to wrap all the query params so that it does not clutter the method argument (as a cleaner solution) when being passed to the service. It will be used for the next ticket when we need to filter by query params